public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -191,7 +185,20 @@ function (angular, _) {
           });
         }
 
-        return this.getDimensionValues(region, namespace, metricName, dimensions);
+        return this.getDimensionValues(region, namespace, metricName, dimensions).then(function(result) {
+          return _.map(result, function(dimensions) {
+            var values = _.chain(dimensions)
+            .sortBy(function(dimension) {
+              return dimension.Name;
+            })
+            .map(function(dimension) {
+              return dimension.Name + '=' + dimension.Value;
```

Comment:
If it is not acceptable, how do you think adding another query expression like following.
dimension_pairs(region,namespace,metrics,...).
This query expression returns the response in "dim_key1=dim_value1,dim_key2=dim_value2..." format.

If it is acceptable, I'll add dimension_key parameter for dimension_values() expression.